crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs

```diff
@@ -0,0 +1,233 @@
+use holo_hash::AnyDhtHash;
+use holochain_cascade::CascadeSource;
+use holochain_types::dht_op::DhtOpType;
+use holochain_zome_types::{
+    record::{Record, SignedActionHashed},
+    Action, action::ActionHashed,
+};
+use std::collections::{HashMap, HashSet};
+
+/// A collection of validation dependencies for the current set of DHT ops requiring validation.
+pub struct ValidationDependencies {
```

Comment:
New data structure for managing dependencies which will be held across calls to the workflow, while they are still needed.